src/os/unix/library_loader_unix.cpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * This file is part of OpenTTD.
+ * OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+ * OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file library_loader_unix.cpp Implementation of the LibraryLoader for Linux / MacOS */
+
+#include "../../stdafx.h"
+
+#include <dlfcn.h>
+
+#include "../../library_loader.h"
+
+#include "../../safeguards.h"
+
+/* Emscripten cannot dynamically load other files. */
+#if defined(__EMSCRIPTEN__)
+
+void *LibraryLoader::OpenLibrary(const std::string &)
+{
+	this->has_error = true;
+	this->error = "Dynamic loading is not supported on this platform.";
+	return nullptr;
+}
+
+void LibraryLoader::CloseLibrary()
+{
+}
+
+void *LibraryLoader::GetSymbol(const std::string &)
+{
+	this->has_error = true;
+	this->error = "Dynamic loading is not supported on this platform.";
```

Comment:
This isn't needed, because `OpenLibrary` already set these in the constructor of the loader.

Reply:
Might not be needed, but does feel more correct. To set the error and the error message. And it doesn't hurt anyone, as you shouldn't be calling this function if OpenLibrary fails.

So a bit pedantic code from my side, but I am going to leave it in :)